archive-xcframework.sh

```diff
@@ -1,83 +1,40 @@
-	-destination 'generic/platform=iOS Simulator'
-
-rm -rf "${OUTPUT_DIR}/iphonesimulator"
-mkdir -p "${OUTPUT_DIR}/iphonesimulator"
-ditto "${DERIVED_DATA_PATH}/Build/Products/Release-iphonesimulator/${FWNAME}.framework" "${OUTPUT_DIR}/iphonesimulator/${FWNAME}.framework"
-rm -rf "${DERIVED_DATA_PATH}"
+    -destination 'generic/platform=iOS Simulator' \
+    -archivePath 'Frameworks/iphonesimulator.xcarchive' \
+    SKIP_INSTALL=NO
 
 # Mac Catalyst
-DERIVED_DATA_PATH=$( mktemp -d )
-xcrun xcodebuild build \
-	$COMMON_SETUP \
+xcrun xcodebuild archive \
+    $COMMON_SETUP \
     -scheme "Double-Conversion" \
-	-derivedDataPath "${DERIVED_DATA_PATH}" \
-	-destination 'platform=macOS,arch=x86_64,variant=Mac Catalyst'
-
-rm -rf "${OUTPUT_DIR}/maccatalyst"
-mkdir -p "${OUTPUT_DIR}/maccatalyst"
-ditto "${DERIVED_DATA_PATH}/Build/Products/Release-maccatalyst/${FWNAME}.framework" "${OUTPUT_DIR}/maccatalyst/${FWNAME}.framework"
-rm -rf "${DERIVED_DATA_PATH}"
-
-#
-
-rm -rf "${BASE_PWD}/Frameworks/iphoneos"
-mkdir -p "${BASE_PWD}/Frameworks/iphoneos"
-ditto "${OUTPUT_DIR}/iphoneos/${FWNAME}.framework" "${BASE_PWD}/Frameworks/iphoneos/${FWNAME}.framework"
-
-rm -rf "${BASE_PWD}/Frameworks/iphonesimulator"
-mkdir -p "${BASE_PWD}/Frameworks/iphonesimulator"
-ditto "${OUTPUT_DIR}/iphonesimulator/${FWNAME}.framework" "${BASE_PWD}/Frameworks/iphonesimulator/${FWNAME}.framework"
-
-rm -rf "${BASE_PWD}/Frameworks/maccatalyst"
-mkdir -p "${BASE_PWD}/Frameworks/maccatalyst"
-ditto "${OUTPUT_DIR}/maccatalyst/${FWNAME}.framework" "${BASE_PWD}/Frameworks/maccatalyst/${FWNAME}.framework"
-
-# XCFramework
-rm -rf "${BASE_PWD}/Frameworks/${FWNAME}.xcframework"
+    -destination 'platform=macOS,arch=x86_64,variant=Mac Catalyst' \
+    -archivePath 'Frameworks/maccatalyst.xcarchive' \
+    SKIP_INSTALL=NO
 
 xcrun xcodebuild -quiet -create-xcframework \
-	-framework "${OUTPUT_DIR}/iphoneos/${FWNAME}.framework" \
-	-framework "${OUTPUT_DIR}/iphonesimulator/${FWNAME}.framework" \
-    -framework "${OUTPUT_DIR}/maccatalyst/${FWNAME}.framework" \
-	-output "${OUTPUT_DIR}/Frameworks/${FWNAME}.xcframework"
-
-# Framework directory structure flattening
-# - Copy the framework in a specific way: de-referencing symbolic links on purpose
-cp -rp "${OUTPUT_DIR}/Frameworks/${FWNAME}.xcframework" "${BASE_PWD}/Frameworks/"
-
-# Catalyst framework directory structure flattening
-# - Remove the catalyst framework "Versions" directory structure after symbolic link de-reference
-rm -rf "${BASE_PWD}/Frameworks/${FWNAME}.xcframework/ios-arm64_x86_64-maccatalyst/${FWNAME}.framework/Versions"
-
-rm -rf ${OUTPUT_DIR}
+    -framework "Frameworks/iphoneos.xcarchive/Products/Library/Frameworks/${FWNAME}.framework" \
+    -framework "Frameworks/iphonesimulator.xcarchive/Products/Library/Frameworks/${FWNAME}.framework" \
+    -framework "Frameworks/maccatalyst.xcarchive/Products/Library/Frameworks/${FWNAME}.framework" \
+    -output "${BASE_PWD}/Frameworks/${FWNAME}.xcframework"
```

Comment:
I'm also very curious why directory flattening is not needed now but was needed prior 🤔 - I'm pretty sure I'm the source of that flattening code from a year or two back and in my experimentation it was literally the only thing that worked. I believe that was on Xcode 12.4 to start with, before M1 existed. Perhaps the tooling changed a bit. Either way if this actually works for both intel and arm64 macs at the same time, fantastic!